src/index/blockfilterindex.cpp

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <map>
+
+#include <index/blockfilterindex.h>
+#include <util/system.h>
+#include <validation.h>
+
+/* The index database stores three items for each block: the disk location of the encoded filter,
+ * its dSHA256 hash, and the header. Those belonging to blocks on the active chain are indexed by
+ * height, and those belonging to blocks that have been reorganized out of the active chain are
+ * indexed by block hash. This ensures that filter data for any block that becomes part of the
```

Comment:
In commit "index: Implement block filter index with write operations." (5964d2f97a221b898dbc477ca8f06ddfbe1d241e)

It'd be good to mention that values and not just the keys are different in these two cases (height keys appear to have values prefixed by the block hash).